packages/bloc_test/lib/src/bloc_test.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:meta/meta.dart';
+import 'package:test/test.dart';
+
+/// Creates a new [bloc]-specific test case with the given [description].
+/// [blocTest] will handle asserting that the [bloc] emits the [expect]ed
+/// states (in order) after the given event is [add]ed.
+/// [blocTest] also handles ensuring that no additional states are emitted
+/// by closing the bloc stream before evaluating the [expect]ation.
+///
+/// [setUp] should be used to all [bloc] initialization and preparation
+/// and must return the [bloc] under test.
+///
+/// [add] is an optional event which will be [add]ed to the [bloc].
+///
+/// [expect] is an `Iterable<State>` which we expect the [bloc]
+/// under test to emit after the provided event is [add]ed.
+///
+/// ```dart
+/// blocTest(
+///   'CounterBloc emits [0, 1] when CounterEvent.increment is added',
+///   setUp: () => CounterBloc(),
```

Comment:
Maybe `build/builder` instead of `setUp`? I "think" it might lead to confusion with the preexisting `setUp` method of the test suite...